### ISSUE TITLE
`number_separator`: allow ignoring certain types of numbers

### ISSUE THREAD

**Issue by rgoldberg** [realm/SwiftLint#5932]
### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Feature or Enhancement Proposal

`number_separator` should be able to be configured to ignore/exclude certain types / typealiases of numbers like:

```yaml
number_separator:
  ignore: [pid_t]
```

So any literal used for a `pid_t` is ignored by this rule.

**Comment by SimplyDanny** [comment]
In which regard would this be helpful? Why would numbers used for `pid_t`s not benefit from separators?

**Comment by rgoldberg** [comment]
There are some typealiases that I'd prefer to see without underscores. Mainly if they're IDs instead of counts. It's just a personal preference. Not the most necessary option, but figured I'd document it here in case others would like it, too.

**Comment by SimplyDanny** [comment]
Note, that this only works well for declarations with initializations such as `let id: ID = 1234` or `let id = ID(123)`. Otherwise there's no way to know the type of a literal to be ignored.

**Comment by rgoldberg** [comment]
A cast or the type of a function parameter can also indicate the type of the literal.

**Comment by SimplyDanny** [comment]
Cast yes, function parameters only for default values. At the call side, there is no way to know the type of the argument, at least not for fast `lint` rules like `number_separator`.
